funk-palace/s3archiver
# Setup initial docker config

Created both a `Dockerfile` for production use and a `docker-compose` for development use. We can modify / trim down later. But this is a good starting point. Also updated the readme.

One note about the docker build, it uses a two-step build stage approach. The first builds the go binary, but because the image contains all of the `golang` and `dep` source code, it's rather large `~150mb`. So instead, the first stage builds the binary, and then the second stage builds a very minimal `alpine` linux distro and copies the go binary on to it. This gets the image size down to like `~15mb`. Only thing I can't seem to do in the docker file is remove the intermediary image and dependent images (i.e. `golang:alpine`). However, a quick `docker image prune` should do the job.

To Test:
1. Try starting up in dev with `docker-compose up -d`. 
2. Go to `localhost:8080/status` and verify you get an ok response.
3. Shut down the dev server with `docker-compose down`.
4. Try building a production image with `docker build -t s3archiver .`.
5. Start running an s3archiver container with `docker run -i -t -p 8080:8080 s3archiver`.
6. Go to `localhost:8080/status` and verify you get an ok response.
